### PR TITLE
Post detail page

### PR DESCRIPTION
**Pull request by ChrisChrisLoLo**
Create post detail page

Adjust margins on home page

Detail page:
Screenshot_20210312_232937

New margins:
Screenshot_20210312_232951

### CODE REVIEW

File: front-end/src/pages/PostDetailPage.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      getPromise = axios.get(`${process.env.REACT_APP_API_URL}/api/posts/${postId}`,
+        {
+          auth: {
+            username: props.loggedInUser.username,
+            password: props.loggedInUser.password
+          }
+        });
+    }
+    else {
+      getPromise = axios.get(`${process.env.REACT_APP_API_URL}/api/posts/${props.location.pathname}`,);
```
Comment (jmmxp) [inline front-end/src/pages/PostDetailPage.tsx:40]: Just to make sure I understand 100%, what's the difference between making the request w/ Basic Auth and w/o?

**Review by jmmxp — APPROVED** [review]
LGTM, just had a quick question

**Comment by raysarinas** [comment]
> Detail page:
> Screenshot_20210312_232937

Do you just get redirected to the stream/homepage after you close the post details page? I guess the background behind the modal in this screenshot looks funny to me? Could we embed the modal into a page so we can still use the navbar while viewing post details? Not sure if it's possible, or if its something we could hold off on until the very end as something extra

File: front-end/src/pages/PostDetailPage.tsx (hunk 1 above)
Comment (ChrisChrisLoLo) [reply front-end/src/pages/PostDetailPage.tsx:40]: Good question :+1:

This logic is to allow for logged-in users and non-logged-in users to attempt to access a post.
I'm assuming that some friends-only posts need user credentials in order to determine if the server should return the post or not

I'm assuming that the server will refuse to send a friends-only post if 'bad' credentials or no credentials get sent to it

There might need to be additional work to get this edge case completed though

**Comment by ChrisChrisLoLo** [comment]
> > Detail page:
> > Screenshot_20210312_232937
> 
> Do you just get redirected to the stream/homepage after you close the post details page? I guess the background behind the modal in this screenshot looks funny to me? Could we embed the modal into a page so we can still use the navbar while viewing post details? Not sure if it's possible, or if its something we could hold off on until the very end as something extra

Valid concern. You do get redirected when closing the modal. It'd be fairly tricky to embed the modal into a page since IIRC React Router only supports rendering individual pages, not individual components that can be added onto pages. 

There might be other tricks we can do that are fairly simple (give the page a nice texture, make the modal shadow darker, anything that can make the plain background less apparent), so it would be nice to have a ticket for this so we can polish it up at the end. Making the component seamless with the homepage could be fairly non-trivial though, or may involve some hacks/code-duplication

File: front-end/src/pages/PostDetailPage.tsx (hunk 1 above)
Comment (jmmxp) [reply front-end/src/pages/PostDetailPage.tsx:40]: Got it, makes sense. Thanks for clarifying!

**Pull request merged**